mozmeao/birdbox
# Dockerize birdbox

This changeset adds Docker support to the project, as a necessary step to getting it deployed.

It uses a muti-stage build similar to Bedrock's -- indeed it's based on a pared-down version of that same Dockerfile.

Things to note that are different from Bedrock or worth consideration in general:

* we use gunicorn with gevent rather than meinheld and greenlet -- this is because we faced wheel-related installation issues while using python 3.11 and config via pyproject.toml [very similar to this](https://stackoverflow.com/questions/76159927/error-could-not-build-wheels-for-greenlet-which-is-required-to-install-pyproje)
* we don't need a mode where we run supervisord in containers
* The docker-compose setup has postgres and redis as backing services 
* The container supports storing assets in Google Cloud Storage if appropriate credentials are available

## To test

To confirm the Docker build is working and that the cloud storage is also fine

* Check out the branch
* Have `just` installed - see README or `brew install just`
* Download the JSON credentials for the GCS service account that @stevejalim linked to when he told you about this pull request. Save them in `/path/to/birdbox/local-credentials` so that the Dockerfile can get them.
* Duplicate ./docker/envfiles/local.env.example to be local.env and fill it with the env vars @stevejalim gave you.
* `docker-compose build assets app release`
* Run the dev app with `docker-compose up app`
* In a separate session, bootstrap the database etc with  `just docker-preflight`
* Add a superuse with `just docker-manage-py createsuperuser`
* Go to http://localhost:8080 and you should see a simple 'Welcome to Wagtail message' because there is no site content in a bare installation
* Go to http://localhost:8080/admin/ and log in to Wagtail with the superuser creds you made earlier
* You'll see an empty CMS - that's fine for now. 
* Let's confirm the GCS bucket works by going to the Images item in the side menu and then hitting the button to add a new image. Upload it -- if it fails wagtail will tell you.
